pkg/nb/router.go

```diff
@@ -22,7 +22,7 @@ import (
 // APIRouter should be able to map noobaa api names to actual addresses
 // See implementations below: APIRouterNodePort, APIRouterPodPort, APIRouterServicePort
 type APIRouter interface {
-	GetAddress(api string) string
+	GetAddress(api string, protocol string) string
```

Comment:
I think protocol should be determined by the router, rather than provided to it.